lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -2835,6 +2854,102 @@ OpFoldResult AtenDivIntOp::fold(FoldAdaptor adaptor) {
   return nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// AtenIndexSelectOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult AtenIndexSelectOp::fold(FoldAdaptor adaptor) {
+  auto self = getSelf();
+  auto index = getIndex();
+  auto selfTy = dyn_cast_or_null<ValueTensorType>(self.getType());
+  auto indexTy = dyn_cast_or_null<ValueTensorType>(index.getType());
+  auto resultTy = dyn_cast_or_null<ValueTensorType>(getType());
+  if (!selfTy || !indexTy || !resultTy)
+    return nullptr;
+
+  if (!selfTy.hasSizes() || !indexTy.hasSizes() || !resultTy.hasSizes())
```

Comment:
Same with the check below, we might as well `||` them together.

Reply:
Ok, will do.

Reply:
done